File: src/models/Volume.ts
export interface Volume {
  id: string;
  comprimento: number;
  largura: number;
  altura: number;
  quantidade: number;
  pesoUnit: number;
}

export function pesoTotal(v: Volume): number {
  return v.quantidade * v.pesoUnit;
}

export function pesoCubadoTotal(v: Volume): number {
  return (v.comprimento * v.largura * v.altura / 6_000) * v.quantidade;
}

export function dimensoesText(v: Volume): string {
  return `${Math.round(v.comprimento)}×${Math.round(v.largura)}×${Math.round(v.altura)}`;
}

File: MedidorCaixas/Models/Volume.swift
import Foundation

struct Volume: Identifiable {
    let id = UUID()
    var comprimento: Double
    var largura: Double
    var altura: Double
    var quantidade: Int
    var pesoUnit: Double

    var pesoTotal: Double { Double(quantidade) * pesoUnit }
    var volumeM3: Double { comprimento * largura * altura / 1_000_000 }
    var pesoCubadoUnit: Double { comprimento * largura * altura / 6_000 }
    var pesoCubadoTotal: Double { pesoCubadoUnit * Double(quantidade) }
    var dimensoesText: String { "\(Int(comprimento))×\(Int(largura))×\(Int(altura))" }
}
